### DISCUSSION TITLE
x-default alternate links and `localePrefix: "always"`

### DISCUSSION THREAD

**Discussion by stefanprobst** [amannn/next-intl#799] [Q&A] [answered]
hi, quick question about the `x-default` alternate link header, which is added out of the box. currently, this defaults to the unprefixed pathname, even when using the `localePrefix: "always"` routing strategy.

i was wondering if prefixing with the default locale would make sense (also, since the [default middleware matcher](https://next-intl-docs.vercel.app/docs/routing/middleware) in the docs does not actually redirect unprefixed pathnames except "/")?

**Comment by amannn** [comment, marked as answer]
Thanks for the great question @stefanprobst! I've added https://github.com/amannn/next-intl/pull/805 to address this.

**Comment by Hadeed-S** [comment]
@amannn X-Default still appears for non-root pages with nextjs 16.0.7, next-intl 4.5.2

i18 config:
`export const routing = defineRouting({
  locales: configs.locales.map((locale) => locale.value), // ['en','ar']
  defaultLocale: defaultLocale, // 'ar'
  localeDetection: false,
  localePrefix: 'always',
`

It also appears at /pathnames route for the example present in the Docs: https://next-intl-example-app-router.vercel.app/en/pathnames

**Reply by amannn** [reply]
Please see here: https://next-intl.dev/blog/next-intl-4-0#other-changes

**Reply by Hadeed-S** [reply]
Understood, Thank you.
